sdk/typescript/src/utils/move-registry.ts

```diff
@@ -0,0 +1,136 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import type { TransactionDataBuilder } from '../transactions/TransactionData.js';
+
+/** The pattern to find an optionally versioned name */
+const NAME_PATTERN = /^([a-z0-9]+(?:-[a-z0-9]+)*)@([a-z0-9]+(?:-[a-z0-9]+)*)(?:\/v(\d+))?$/;
+const NAME_SEPARATOR = '@';
+
+export type NameResolutionRequest = {
+	id: number;
+	type: 'package' | 'moveType';
+	name: string;
+};
+
+export type NamedPackagesPluginCache = {
```

Comment:
these types should be exported along side the plugin, rather than needing to be imported separately from utils